Completed Input Sanitization, Resolving #83 and #84

BellJrDev opened the pull request:
My env variables are not setup, so all of the const runner stuff is a problem for me. But commenting that part out, the code works just fine on my end. I may or may not need to put documentation in for the class. I am not documenting types, nor do I expect the rest of the team to do so.  It's not the prettiest, but it works and it's a rather lightweight solution. John please don't hurt me.

BellJrDev committed:
Updated Input.ts documentation

BellJrDev committed:
Merge branch 'bell' into sanitize

BellJrDev committed:
Wrote GitHub sanitization. Need to get repo name & owner from npm.

BellJrDev committed:
Updated gitIgnore

BellJrDev committed:
Merge branch 'bell' into sanitize

BellJrDev committed:
Completed sanitization. ProvideURLsForQuerying does as name suggests with tokenized URLs

BellJrDev committed:
Merge branch 'dev' into sanitize

j-leidy commented on MVP/src/index.ts:
```diff
@@ -18,8 +18,10 @@ import { mapGQLResultToRepos } from './Processors/gqlProcessor';
-console.log(contents);
+let cleanUrls = InputHandling.ProvideURLsForQuerying(exampleFilepath);
+console.log(cleanUrls.github_URLs);
+console.log(cleanUrls.npm_URLs);
```
![image](https://github.com/user-attachments/assets/96979b61-5b3b-4b82-bbc4-d35d678980a2)
Npm urls appear to be broken currently, showing package for packageName

j-leidy replied:
it's also worth mentioning that we should parse out '\n' and '\r'

BellJrDev committed:
Fixed bug where packageName was always package

BellJrDev committed:
:
Merge branch 'sanitize' of https://github.com/BellDorian/SoftwareEng_Fall2024 into sanitize

j-leidy commented on MVP/src/Input/Input.ts:
```diff
@@ -14,14 +86,130 @@ import { readFileSync } from 'fs';
+    for (var i = 0; i < size; i++) {
+        try {
+            let url = rawUrls[i];
+            if (url.length < 11) {
```
why should the url be less than 11 characters?

BellJrDev committed:
Function ProvideURLsForQuerying does as name suggests. Sending updated index.ts and Input.ts
